src/s3fs.cpp

```diff
@@ -2336,6 +2336,12 @@ static int s3fs_statfs(const char* _path, struct statvfs* stbuf)
     stbuf->f_bfree  = stbuf->f_blocks;
     stbuf->f_bavail = stbuf->f_blocks;
     stbuf->f_namemax = NAME_MAX;
+#ifdef __MSYS__
+    // WinFsp resolves the free space from f_bfree * f_frsize, and the total space from f_blocks * f_frsize (in bytes).
+    stbuf->f_frsize = stbuf->f_bsize;
+    stbuf->f_blocks = INT32_MAX;
+    stbuf->f_bfree = INT32_MAX;
+#endif
```

Comment:
Should this have `#else` to avoid setting `f_blocks` and `f_bfree` more than once?  You need to shuffle the existing code to do this.

Reply:
Fixed at 1e12274.